process-dir.js

```diff
@@ -164,16 +166,26 @@ async function replacePackage (dir, file, files) {
 }
 
 async function process (dir) {
+  let ignorePatterns = []
+  let removePatterns = []
+
   let npmignorePath = join(dir, '.npmignore')
-  let ignore = []
   if (fs.existsSync(npmignorePath)) {
-    ignore = await readFile(npmignorePath)
-    ignore = ignore.toString().split('\n').filter(i => !!i).map(i => {
+    removePatterns = (await readFile(npmignorePath)).toString()
+    removePatterns = removePatterns.split('\n').filter(i => !!i).map(i => {
       return i.endsWith(sep) ? i.slice(0, -1) : i
     })
   }
 
-  ignore.push('**/*.test.js', '**/*.spec.js')
+  removePatterns.push('**/*.test.js', '**/*.spec.js')
+
+  let filesForRemove = await globby(removePatterns, { cwd: dir })
+
+  filesForRemove.forEach(async fileForRemove => {
+    await rimraf(join(dir, fileForRemove))
```

Comment:
This `await` will block only `async fileForRemove => {` callback, but not `process` function.

As a result, `removeEmptyDirectories` will be called before files removing.

You need:

```js
await Promise.all(filesForRemove.map(async file => rimraf(join(dir, fileForRemove)))
```